tests/functional/src/main/java/org/mule/functional/classloading/isolation/maven/MavenArtifactToClassPathURLResolver.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.classloading.isolation.maven;
+
+import java.net.URL;
+import java.util.List;
+
+/**
+ * Resolves a {@link MavenArtifact} by selecting from the list of URLs the one that matches
+ *
+ * @since 4.0
+ */
+public interface MavenArtifactToClassPathURLResolver
+{
+
+    /**
+     * @param artifact to be used in order to find the {@link URL} in list of urls
+     * @param urls a list of {@link URL}, most of the cases the one provided by the classpath
```

Comment:
"most of the cases the one provided by the classpath": in which cases the URL won't be provided by the classpath?